doc/md/language-manual.md

```diff
@@ -1654,6 +1654,24 @@ Such an object literal, sometimes called a *record*, is equivalent to the object
 
 Object expressions support *punning* for concision. A punned field `<id>` is shorthand for `<id> = <id>`; Similarly, a typed, punned field `<id> : <typ>` is short-hand for `<id> = <id> : <typ>`. Both associate the field named `<id>` with the value of the identifier `<id>`.
 
+#### Object reuse and extension
+
+Objects can be combined and/or extended using the `and` and `with` keywords.
+
+A record expression `{ <exp_post> (and <exp_post>)* with <exp-field>;* }` merges the objects (or modules) specified as *bases* which can be
+postfixed expressions, and augments the result to also contain the specified fields. The field list can be omitted when at least two bases appear and none have common field labels.
```

Comment:
```suggestion

```
(For better or worse, the manual deals only with abstract syntax, not concrete syntax, so I'd skip the distinction between <exp_post> and <exp>).
```suggestion
A record expression `{ <exp> (and <exp>)* (with <exp-field>;+)? }` merges the objects (or modules) specified as *base* expressions, and augments the result to also contain the specified fields. The `with <exp-field>;+` clause can be omitted when at least two bases appear and none have common field labels.
```

Reply:
The `{ <exp> (and <exp>)* (with <exp-field>;+)? }` should also be added to the figure defining expression syntax earlier in the document. (Search for `<exp> ::=`  I guess).

It would be nice if you could describe the typing rule and evaluation rules for the expression, but I think that might be quite a mouthfull. Perhaps I could have a go later sometime.